magento/pwa-studio
# SW PDP error fix for IMAGE_OPTIMIZING_ORIGIN=backend

## Description

When **IMAGE_OPTIMIZING_ORIGIN** is set to **backend**, **resourceUrl** function of **venia-drivers** package returns an absolute URL instead of a relative URL in **useProductImageCarousel**. SW would not accept any origin other than the origin of the app that created it due to CORS restrictions.

Fixed this by stripping away the backend origin set by **resourceUrl** before sending it to the SW in case of **IMAGE_OPTIMIZING_ORIGIN** set to **backend**.

## Related Issue
Closes [PWA-392](https://jira.corp.magento.com/browse/PWA-392)

### Verification Steps
1. Set `IMAGE_OPTIMIZING_ORIGIN` to `backend` in `.env` file.
2. Build and stage venia using `yarn build && yarn stage:venia`.
3. Go to any PDP page.
4. Should not see any errors in the console.

## Checklist
* Should not change the app's behavior in watch mode.
* No app functionality has been changed.
